apis/nucleus/src/components/listbox/ListBoxInline.jsx

```diff
@@ -211,24 +211,26 @@ function ListBoxInline({ options, layout }) {
       {toolbar && (
         <Grid item container style={{ padding: theme.spacing(1) }} wrap="nowrap">
           <Grid item container height={headerHeight} wrap="nowrap">
-            <Grid item sx={{ display: 'flex', alignItems: 'center', width: searchIconWidth }}>
-              {isLocked ? (
-                <IconButton tabIndex={-1} onClick={unlock} disabled={!isLocked} size="large">
-                  <Lock title={translator.get('Listbox.Unlock')} style={{ fontSize: '12px' }} />
-                </IconButton>
-              ) : (
-                searchEnabled !== false && (
-                  <IconButton
-                    onClick={onShowSearch}
-                    tabIndex={-1}
-                    title={translator.get('Listbox.Search')}
-                    size="large"
-                  >
-                    <SearchIcon style={{ fontSize: '12px' }} />
+            {(isLocked || (searchEnabled !== false && !constraints.active)) && (
+              <Grid item sx={{ display: 'flex', alignItems: 'center', width: searchIconWidth }}>
+                {isLocked ? (
+                  <IconButton tabIndex={-1} onClick={unlock} disabled={!isLocked} size="large">
```

Comment:
maybe also disable unlock if constraints.active (or constraints.select)

Reply:
Updated it!